sound/soc/intel/boards/bxt_da7219_max98357a.c

```diff
@@ -617,12 +617,15 @@ static int bxt_card_late_probe(struct snd_soc_card *card)
 		snd_soc_dapm_add_routes(&card->dapm, broxton_map,
 					ARRAY_SIZE(broxton_map));
 
-	pcm = list_first_entry(&ctx->hdmi_pcm_list, struct bxt_hdmi_pcm,
-			       head);
-	component = pcm->codec_dai->component;
+	if (list_empty(&ctx->hdmi_pcm_list))
+		return -EINVAL;
```

Comment:
This is basicly hardcoded to the machine drivers DAI list, so in practise can't happen. But hmm, especially if this is pulled to a common function, better to have the check still.